skimage/filters/rank/generic.py

```diff
@@ -146,6 +150,57 @@ def _preprocess_input(image, selem=None, out=None, mask=None, out_dtype=None,
     return image, selem, out, mask, n_bins
 
 
+def _handle_input_3D(image, selem, out, mask, out_dtype=None, pixel_size=1):
+
+    check_nD(image, [2, 3])
```

Comment:
why 2? shouldn't it be always 3D in this function?

Reply:
I believe it should be only 3. @ryanlu41 ?

Reply:
yes, i think that might be left over from when I tried to do 2D and 3D in one function

Reply:
fixed